daemons/ipa-slapi-plugins/ipa-extdom-extop/ipa_extdom_cmocka_tests.c

```diff
@@ -655,6 +645,6 @@ int main(int argc, const char *argv[])
         cmocka_unit_test(test_decode),
     };
 
-    assert_non_null(original_fopen);
+    skip_tests = (unshare(CLONE_NEWUSER) == -1);
```

Comment:
Is there a reason you don't pass CLONE_NEWUSER to clone rather than using unshare? The man page for unshare(2) says it has the same effect and all the mechanism around unshare would be unnecessary.

Reply:
Because it doesn't work in Koji while working locally, I tried. 
Unfortunately, it is not usable there: 
https://kojipkgs.fedoraproject.org//work/tasks/6877/48596877/build.log

```
make  extdom_cmocka_tests
make[4]: Entering directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
gcc -DHAVE_CONFIG_H -I. -I../../..  -I. -I./../common -DPREFIX=\""/usr"\" -DBINDIR=\""/usr/bin"\" -DLIBDIR=\""/usr/lib64"\" -DLIBEXECDIR=\""/usr/libexec"\" -DDATADIR=\""/usr/share"\" -DUSE_OPENLDAP -I/usr/include/dirsrv  -I/usr/include/nspr4        -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D__STDC_WANT_LIB_EXT1__=1 -D_DEFAULT_SOURCE=1 -D_POSIX_C_SOURCE=200809L -Werror=implicit-function-declaration -c -o extdom_cmocka_tests-ipa_extdom_cmocka_tests.o `test -f 'ipa_extdom_cmocka_tests.c' || echo './'`ipa_extdom_cmocka_tests.c
gcc -DHAVE_CONFIG_H -I. -I../../..  -I. -I./../common -DPREFIX=\""/usr"\" -DBINDIR=\""/usr/bin"\" -DLIBDIR=\""/usr/lib64"\" -DLIBEXECDIR=\""/usr/libexec"\" -DDATADIR=\""/usr/share"\" -DUSE_OPENLDAP -I/usr/include/dirsrv  -I/usr/include/nspr4        -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D__STDC_WANT_LIB_EXT1__=1 -D_DEFAULT_SOURCE=1 -D_POSIX_C_SOURCE=200809L -Werror=implicit-function-declaration -c -o extdom_cmocka_tests-ipa_extdom_common.o `test -f 'ipa_extdom_common.c' || echo './'`ipa_extdom_common.c
gcc -DHAVE_CONFIG_H -I. -I../../..  -I. -I./../common -DPREFIX=\""/usr"\" -DBINDIR=\""/usr/bin"\" -DLIBDIR=\""/usr/lib64"\" -DLIBEXECDIR=\""/usr/libexec"\" -DDATADIR=\""/usr/share"\" -DUSE_OPENLDAP -I/usr/include/dirsrv  -I/usr/include/nspr4        -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D__STDC_WANT_LIB_EXT1__=1 -D_DEFAULT_SOURCE=1 -D_POSIX_C_SOURCE=200809L -Werror=implicit-function-declaration -c -o extdom_cmocka_tests-back_extdom_nss_sss.o `test -f 'back_extdom_nss_sss.c' || echo './'`back_extdom_nss_sss.c
/bin/sh ../../../libtool  --tag=CC   --mode=link gcc  -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D__STDC_WANT_LIB_EXT1__=1 -D_DEFAULT_SOURCE=1 -D_POSIX_C_SOURCE=200809L -Werror=implicit-function-declaration -rpath /usr/lib64/dirsrv   -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o extdom_cmocka_tests extdom_cmocka_tests-ipa_extdom_cmocka_tests.o extdom_cmocka_tests-ipa_extdom_common.o extdom_cmocka_tests-back_extdom_nss_sss.o  -lcmocka  -lldap_r -llber -L/usr/lib64/dirsrv -lslapd  -lsss_nss_idmap  -ldl  
libtool: link: gcc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D__STDC_WANT_LIB_EXT1__=1 -D_DEFAULT_SOURCE=1 -D_POSIX_C_SOURCE=200809L -Werror=implicit-function-declaration -Wl,-z -Wl,relro -Wl,--as-needed -Wl,-z -Wl,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o extdom_cmocka_tests extdom_cmocka_tests-ipa_extdom_cmocka_tests.o extdom_cmocka_tests-ipa_extdom_common.o extdom_cmocka_tests-back_extdom_nss_sss.o  -lcmocka -lldap_r -llber -L/usr/lib64/dirsrv -lslapd -lsss_nss_idmap -ldl -Wl,-rpath -Wl,/usr/lib64/dirsrv
make[4]: Leaving directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
make  check-TESTS
make[4]: Entering directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
make[5]: Entering directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
FAIL: extdom_cmocka_tests
==============================================================================
   freeipa 4.8.7: daemons/ipa-slapi-plugins/ipa-extdom-extop/test-suite.log
==============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
.. contents:: :depth: 2
FAIL: extdom_cmocka_tests
=========================
FAIL extdom_cmocka_tests (exit status: 2)
============================================================================
Testsuite summary for freeipa 4.8.7
============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See daemons/ipa-slapi-plugins/ipa-extdom-extop/test-suite.log
Please report to https://hosted.fedoraproject.org/projects/freeipa/newticket
============================================================================
make[5]: Leaving directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
make[4]: Leaving directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
make[5]: *** [Makefile:983: test-suite.log] Error 1
make[4]: *** [Makefile:1091: check-TESTS] Error 2
make[3]: Leaving directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins/ipa-extdom-extop'
make[3]: *** [Makefile:1165: check-am] Error 2
make[2]: *** [Makefile:491: check-recursive] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/freeipa-4.8.7/daemons/ipa-slapi-plugins'
make[1]: *** [Makefile:485: check-recursive] Error 1
make[1]: Leaving directory '/builddir/build/BUILD/freeipa-4.8.7/daemons'
make: *** [Makefile:679: check-recursive] Error 1
+ cat daemons/ipa-slapi-plugins/ipa-extdom-extop/extdom_cmocka_tests.log
FAIL extdom_cmocka_tests (exit status: 2)
+ exit 1
error: Bad exit status from /var/tmp/rpm-tmp.Zoxpby (%check)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.Zoxpby (%check)
Child return code was: 1
EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mockbuild/trace_decorator.py", line 93, in trace
    result = func(*args, **kw)
  File "/usr/lib/python3.8/site-packages/mockbuild/util.py", line 776, in do_with_status
    raise exception.Error("Command failed: \n # %s\n%s" % (command, output), child.returncode)
mockbuild.exception.Error: Command failed: 
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/freeipa.spec

```

Reply:
The exist status 2 corresponds to the following snippet in my test code:
```
    pid = clone(main_test, stackTop, CLONE_NEWUSER | SIGCHLD, NULL);
    if (pid == -1) {
        exit(EXIT_FAILURE + 1);
    }

```

Koji build environments do not allow to execute `clone()` with `CLONE_NEWUSER`.